core/src/main/java/org/infinispan/transaction/impl/TransactionTable.java

```diff
@@ -390,7 +390,12 @@ private RemoteTransaction getOrCreateRemoteTransaction(GlobalTransaction globalT
             if (topologyId < minTxTopologyId) {
                if (trace)
                   log.tracef("Changing minimum topology ID from %d to %d", minTxTopologyId, topologyId);
-               minTxTopologyId = topologyId;
+               minTopologyRecalculationLock.lock();
```

Comment:
I'm not sure if it makes much sense to lock here. at lest, include the `if (topologyId < minTxTopologyId)` in the locked section.